src/Jogo/SpriteImporter.cs

```diff
@@ -0,0 +1,45 @@
+
+using Raylib_cs;
+
+/// <summary>
+/// Responsible for importing the sprites into the game via the sprite atlas.
+/// </summary>
+public class SpriteImporter {
+    private const string _AtlasPath = "Assets/sprite-atlas.png";
+    private const int _SpriteWidth = 16;
+    private const int _SpriteHeight = 16;
+
+    private static SpriteImporter instance = new SpriteImporter();
+
+    private Image _atlas;
+
+    /// <summary>
+    /// The instance of the <see cref="SpriteImporter"/> class and loads necessary sprites.
+    /// </summary>
+    /// <param name="position">The initial position of the player entity.</param>
+    private SpriteImporter() {
+        _atlas = Raylib.LoadImage(_AtlasPath);
+    }
+
+    /// <summary>
+    /// Method <c>GetSprite</c> is responsible for restrieving the desired slice of the atlas.
+    /// </summary>
+    /// <param name="i">Sprite row</param>
+    /// <param name="j">Sprite column</param>
+    /// <returns>Texture2D slice of the atlas in the desired position.</returns>
+    public static Texture2D GetSprite(int i, int j) {
+        Rectangle imageBounds = new Rectangle(j * _SpriteWidth, i * _SpriteHeight, _SpriteWidth, _SpriteHeight);
```

Comment:
Cria uma classe tambem singleton, "Atlas" e mantem nela uma lista de `Texture`, e nela ser possivel obter o sprite pela posição tambem, porque é importante que a cada impressão não seja dado load da imagem de dado unload a cada frame.

Reply:
A ideia não é que a cada impressão seja dado load, e sim que seja carregado como readonly static pelas classes q usarem.

Dá pra também manter a lista de texturas no SpriteImporter, ent ele carregaria todas de uma vez e só retornaria o objeto dentro do array, o que você acha?

Reply:
Então não é um SpriteImporter, é um SpriteManager ou um Atlas por si só, e é ineficiente que cada Entidade carregue um texture2d por causa que isso repetir na memoria ram n vezes o mesmo sprite, e são muitas entidades, e também, o que to descrevendo é basicamente implementar um flyweight

https://refactoring.guru/design-patterns/flyweight

Reply:
Concordo em transformar em outro pattern para concordar com os requerimentos do enunciado, porém eu sugeri que cada objeto importasse apenas uma vez seu sprite como ```readonly static```.

Vou tentar fazer as alterações necessárias para seguir com a sua ideia, pois com ela posso definir uma função de renderização padrão para os sprites (Que seria um wrapper para a função de DrawTexture da Raylib, pois há opções que não são necessárias nela).

Reply:
O problema está no "Cada objeto importar uma vez seu sprite", por exemplo, no mock tem mais de 10paredes, ou seja a parede seria importada desnecessariamente